### DISCUSSION TITLE
Cant get a crop of image

### DISCUSSION THREAD

**Discussion by AndreyMaslow** [nothings/stb#1358] [Q&A]
Hello! I'm trying to get a crop of image using  `stbir_resize_region` and get the assert : 
 `stb_testing: /home/andrey/CLionProjects/stb_testing/3dparty/stb/stb_image_resize.h:1140: stbir__normalize_downsample_coefficients: Assertion total > 0.9f' failed.`
 
What i'm doing wrong or there is another method for image cropping? Please Help! 
The full source : 
 
    int req_height  = 200;
    int req_width   = 200;
    int req_channel = 3;
    
    int height  = 0;
    int width   = 0;
    int channel = 0;

    unsigned char* image_data = stbi_load("/home/andrey/OModule/face_with_landmarks_drawn.jpeg", &width, &height, &channel, req_channel);
    if (image_data == NULL) {
        printf("load image failed!\n");
        return -1;
    }

    if (width != req_width || height != req_height) {
        unsigned char* image_resized = (unsigned char*)STBI_MALLOC(req_width * req_height * req_channel);
        if (!image_resized) {
            printf("malloc image failed!\n");
            STBI_FREE(image_data);
            return -1;
        }

        printf(" h = %d, w = %d, ch = %d\n", width, height, channel);
        
        stbir_resize_region(image_data, width, height, 0, image_resized,
                            req_width, req_height, 0,
                            STBIR_TYPE_UINT8, req_channel, STBIR_ALPHA_CHANNEL_NONE,
                            0, STBIR_EDGE_WRAP, STBIR_EDGE_WRAP, STBIR_FILTER_BOX,
                            STBIR_FILTER_BOX, STBIR_COLORSPACE_LINEAR, NULL, 0, 0, 700, 825);

        if (stbir_resize_uint8(image_data, width, height, 0, image_resized, req_width, req_height, 0, channel) != 1) {
            printf("resize image failed!\n");
            STBI_FREE(image_data);
            return -1;
        }
        STBI_FREE(image_data);

        stbi_write_jpg("/home/andrey/OModule/face_with_landmarks_resized_crop.jpeg", req_width, req_height, req_channel, image_resized, 100);
    }
`

**Comment by nothings** [comment]
what are width and height for this image?

**Reply by AndreyMaslow** [reply]
![face_with_landmarks_drawn](https://user-images.githubusercontent.com/39959517/185619826-75236bbf-4ebc-4e43-86bc-71965013513e.jpeg)

This is the original image. The height is 875, the width is 700

**Reply by AndreyMaslow** [reply]
i added crop by hand, but i'm not sure that it's a efficient way 

```
void get_image_crop_rgb(const uint8_t* input_pixels, int input_w, int crop_x, int crop_y, uint8_t* output_pixels, int output_w, int output_h) {
    int i, j;
    int ch_num = 3;
    int r_channel = 1;
    int g_channel = 2;
    int b_channel = 3;
    for (i = 0; i < output_h; i++) {
        for (j = 0; j < output_w; j++) {
            output_pixels[(i * output_w + j) * ch_num + r_channel] = input_pixels[((crop_y + i) * input_w + crop_x + j) * ch_num + r_channel];
            output_pixels[(i * output_w + j) * ch_num + g_channel] = input_pixels[((crop_y + i) * input_w + crop_x + j) * ch_num + g_channel];
            output_pixels[(i * output_w + j) * ch_num + b_channel] = input_pixels[((crop_y + i) * input_w + crop_x + j) * ch_num + b_channel];
        }
    }
} 
```
